charts/opentelemetry-collector/templates/daemonset.yaml

```diff
@@ -10,6 +10,17 @@ spec:
     matchLabels:
       {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
       component: agent-collector
+  updateStrategy:
+    rollingUpdate:
+      {{- if .Values.agentCollector.rollingUpdate.maxSurge }}
+      maxSurge: {{ .Values.agentCollector.rollingUpdate.maxSurge }}
+      {{- end }}
+      {{- if .Values.agentCollector.rollingUpdate.maxUnavailable }}
+      maxUnavailable: {{ .Values.agentCollector.rollingUpdate.maxUnavailable }}
+      {{- else }}
+      maxUnavailable: 1
+      {{- end }}
+    type: {{ .Values.agentCollector.updateStrategy }}
```

Comment:
What happens if `updateStrategy` is set to `OnDelete`? My gut says that the `rollingUpdate` section should only be rendered if `updateStrategy` is set to `RollingUpdate`

Reply:
You're right @TylerHelmuth . Thank you, fixed.